iocage/lib/ioc_json.py

```diff
@@ -101,26 +97,37 @@ def json_convert_from_zfs(self, uuid, skip=False):
 
         if not skip:
             # Set jailed=off and move the jailed dataset.
-            iocage.lib.ioc_common.checkoutput(["zfs", "set", "jailed=off",
-                                               f"{dataset}/root/data"],
-                                              stderr=su.PIPE)
-            iocage.lib.ioc_common.checkoutput(["zfs", "rename", "-f",
-                                               f"{dataset}/root/data",
-                                               f"{dataset}/data"],
-                                              stderr=su.PIPE)
-            iocage.lib.ioc_common.checkoutput(["zfs", "set",
-                                               f"{jail_zfs_prop}=iocage/jails/"
-                                               f"{uuid}/data",
-                                               f"{dataset}/data"],
-                                              stderr=su.PIPE)
-            iocage.lib.ioc_common.checkoutput(["zfs", "set", "jailed=on",
-                                               f"{dataset}/data"],
-                                              stderr=su.PIPE)
+            self.zfs_set_property(f"{dataset}/root/data", "jailed", "off")
+            self.zfs.get_dataset(f"{dataset}/root/data").rename(f"{dataset}/data")
+            self.zfs_set_property(f"{dataset}/data", jail_zfs_prop, f"iocage/jails/{uuid}/data")
+            self.zfs_set_property(f"{dataset}/data", "jailed", "on")
 
         key_and_value["jail_zfs_dataset"] = f"iocage/jails/{uuid}/data"
 
         self.json_write(key_and_value)
 
+    def _zfs_get_properties(self, identifier):
+        if "/" in identifier:
+            dataset = self.zfs.get_dataset(identifier)
+            return dataset.properties
+        else:
+            pool = self.zfs.get(identifier)
+            return pool.root_dataset.properties
+
+    def zfs_get_property(self, identifier, key):
+        try:
+            return self._zfs_get_properties(identifier)[key].value
+        except:
+            return ""
```

Comment:
Indeed. It shoud be `-` to match with the default output of `zfs get -H -o value unknown:property`. This way it exactly matches the previous behavior.

And no, we don't want to raise an error. For example we read properties that we check for a potential migration. They don't exist when the dataset does not need an upgrade.